src/PrestaShopBundle/Form/Admin/Sell/Order/CartSummaryType.php

```diff
@@ -48,16 +54,37 @@ class CartSummaryType extends AbstractType
      */
     private $paymentModulesChoiceProvider;
 
+    /**
+     * @var int
+     */
+    private $defaultOrderState;
+
+    /**
+     * @var array
+     */
+    private $paymentOrderStates = [];
+
     /**
      * @param FormChoiceProviderInterface $orderStatesChoiceProvider
      * @param FormChoiceProviderInterface $paymentModulesChoiceProvider
+     * @param Configuration $configuration
      */
     public function __construct(
         FormChoiceProviderInterface $orderStatesChoiceProvider,
-        FormChoiceProviderInterface $paymentModulesChoiceProvider
+        FormChoiceProviderInterface $paymentModulesChoiceProvider,
+        Configuration $configuration
     ) {
+        $this->configuration = $configuration;
         $this->orderStatesChoiceProvider = $orderStatesChoiceProvider;
         $this->paymentModulesChoiceProvider = $paymentModulesChoiceProvider;
+        $this->defaultOrderState = (int) $this->configuration->get('PS_OS_PAYMENT');
+        $this->paymentOrderStates = [
+            'ps_checkpayment' => (int) $this->configuration->get('PS_OS_CHEQUE'),
```

Comment:
These are hardcoded modules ? 😮

Can you explain why you need these informations ?

Reply:
It's the default order state relative to payment module.

Legacy : https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminOrdersController.php#L245

Reply:
Waho 😅 

Question: is it OK that in legacy it was named `$defaults_order_state` but you named it `paymentOrderStates` (and used `defaultOrderState` for another thing).

Suggestion, not mandatory: add a comment to explain this, because it's not obvious and if I was surprised by it, I think other people will be surprised (and have the same question) too. So the comment will carry the answer.

Reply:
Renamed variable ;)

Reply:
This should be dynamic based on order_state and modules..
In `ps_order_state` you have a `module_name` column that will allow you to match the module, and the waiting status seems to be the one whose template is `payment` in the `ps_order_state_lang`

So you could add add the `FormChoiceAttributeProviderInterface` to `prestashop.adapter.form.choice_provider.installed_payment_modules` which would return a dynamic array `['module_name' => pending_order_state_id]`